src/misc/like.ts

```diff
@@ -0,0 +1,243 @@
+import { DatabaseEntry } from '../common/datas/database';
+import { dataType, dataHolder } from '../common/datas/holder';
+import { dataView } from '../common/datas/view';
+import * as yup from 'yup';
+import { Bukkit, OfflinePlayer } from 'org.bukkit';
+import { UUID } from 'java.util';
+import { isAdminAccount } from '../common/helpers/player';
+import { Player } from 'org.bukkit.entity';
+import { CommandSender } from 'org.bukkit.command';
+
+const playerLikesDataType = dataType('playerLikesData', {
+  playerUuidList: yup.array().of(yup.string().required()),
+});
+
+const playerLikesType = dataType('playerLikes', {
+  count: yup.number(),
+  lastAutoRemoveDate: yup.number(),
+  cooldownEnds: yup.number(),
+});
+
+const view = dataView(
+  playerLikesDataType,
+  new DatabaseEntry('player-likes', 'player-likes-key'),
+);
+
+if (!view.playerUuidList) view.playerUuidList = [];
+
+/**
+ * Get shape object of top 10 most liked playerUuidList
+ */
+
+function getLiked() {
+  if (!view.playerUuidList || !view.playerUuidList) return [];
+  return sortLikeList(view.playerUuidList);
+}
+
+/**
+ * Sort the given list by likes
+ */
+function sortLikeList(array: string[]): string[] {
+  return array.sort(
+    (a, b) =>
+      getLikes(uuidToOfflinePlayer(b)) - getLikes(uuidToOfflinePlayer(a)),
+  );
+}
+
+/*
+ * Get likes of player
+ */
+function getLikes(player: OfflinePlayer): number {
+  const playerView = dataView(playerLikesType, dataHolder(player));
```

Comment:
Passing player directly to dataView should work, now that you've added OfflinePlayer to allowed types.